src/Microsoft.Health.Fhir.SqlServer/Features/Operations/Import/DataGenerator/TokenTextSearchParamsTableBulkCopyDataGenerator.cs

```diff
@@ -42,7 +46,7 @@ internal override void FillDataTable(DataTable table, SqlBulkCopyDataWrapper inp
 
             IEnumerable<BulkTokenTextTableTypeV1Row> searchParams = _searchParamGenerator.GenerateRows(new ResourceWrapper[] { input.Resource });
 
-            foreach (BulkTokenTextTableTypeV1Row searchParam in searchParams)
+            foreach (BulkTokenTextTableTypeV1Row searchParam in Distinct(searchParams))
```

Comment:
>Distinct [](http://example.com/codeflow?start=64&length=8)

I doubt this problem unique for text only.

Can we change `BulkSearchParameterRowGenerator` to Genereate unique rows and do distinct on ISearchValue objects instead of doing it here?

Reply:
I also don't like how responsibility of knowing what is distinct means leaks into here. We already define GetHashCode and Equals to `ISearchValues` we shouldn't define it again here.

Reply:
One more thing...
Richa recently merge code which make sure `ISearchIndexer` extracts unique values for resource (before it could return duplicates). Does problem still exist with new changes? If yes, can you point to place where we actually get duplicate records? 
Uniqueness of data should be solved once and on bottom level rather than solving it on higher levels in multiple places.

Reply:
@Ivanidzo4ka good idea, agree that we should distinct at central place. @apurvabhaleMS do you have any idea on this?

Reply:
For ISearchIndexer distinct the problem is tokentext contains code and text, and only text in tokentext table. That's possible text is same but code is different for search params.